elastic/kibana
# Check for indices before enabling get search profile in serverless

Closes [#195342](https://github.com/elastic/kibana/issues/195342)

## Summary
In serverless, the default query for the search profiler fails if there is not indices. For avoiding this error, when there are no indices present, this PR disabled the "Profile" button and add a tooltip explaining why it is disabled.

### New strings
This is the tooltip for string verification @kibana-docs [[Code](https://github.com/elastic/kibana/pull/201630/commits/5832a76683ad0cf55558655ca5981d623f344b72?diff=unified&w=0#diff-bf48cd9834b39a2a1634680225fc63c9a4ddb3ca881d9120f648006ad0277046R154-R1552?diff=unified&w=0#diff-bf48cd9834b39a2a1634680225fc63c9a4ddb3ca881d9120f648006ad0277046R155)]:
<img width="460" alt="Screenshot 2024-11-25 at 16 15 08" src="https://github.com/user-attachments/assets/a3395cfb-fc5e-4c22-9dd8-954a60fd1b5d">

### How to test
* Run Kibana in serverless
* Go to Index Management and verify you haven't indices (or delete them if you do have indices).
* Go to Dev Tools and click the Search Profiler tab. Verify that the button is disabled and the tooltip displayed if you hover over it.
* Go back to Index Management and create one or more indices.
* Go back to Dev Tools > Search Profiler. Now the button should be enabled and the profile should be created if you click it. 

### Demo

https://github.com/user-attachments/assets/9bda072e-7897-4418-a906-14807e736c44


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.
